src/Text/XML/ParentAttributes.hs

```diff
@@ -0,0 +1,39 @@
+module Text.XML.ParentAttributes
+  ( ToXmlParentAttributes(..)
+  , ToXmlAttribute(..)
+  , toXmlAttributeIntegral
+  ) where
+
+import Data.Text as T.S
+import Data.Text.Lazy as T.L
+import Data.Text.Lazy.Builder as T.L
+import Data.Text.Lazy.Builder.Int as T.L
+import Numeric.Natural
+import Text.XML
+
+class ToXmlParentAttributes a where
+  toXmlParentAttributes :: a -> [(Name, T.S.Text)]
+
+instance {-# OVERLAPPABLE #-} ToXmlParentAttributes a where
```

Comment:
I wrote a comment that describes the motivation behind this instance and how it's used, with an example. Does it help?